tests/console/repo_package_install.pm

```diff
@@ -14,23 +14,53 @@ use base "consoletest";
 use strict;
 use testapi;
 use utils;
-use version_utils qw(sle_version_at_least is_jeos);
+use version_utils qw(is_sle is_jeos);
 
 my %packages = (
-    # On JeOS Salt is present in the default image
     salt => {
         repo      => 'Basesystem',
-        installed => is_jeos() ? 1 : 0
-    });
+        installed => is_jeos() ? 1 : 0,       # On JeOS Salt is present in the default image
+        condition => sub { is_sle('15+') },
+    },
+    'update-test-feature' => {                # See poo#36451
+        repo      => 'SLES12',
+        installed => 0,
+        available => sub { get_var('BETA') },
+        condition => sub { is_sle('<15') },
+    },
+);
+# Define more packages with the same set of expectations
+$packages{'update-test-interactive'} = $packages{'update-test-feature'};
+$packages{'update-test-security'}    = $packages{'update-test-feature'};
+$packages{'update-test-trival'}      = $packages{'update-test-feature'};
 
 sub run {
     select_console 'root-console';
-    # for now only test for SLE>=15
-    return unless sle_version_at_least('15');
+    my $errors = '';    # Variable to accumulate errors
     for my $package (keys %packages) {
+        # Skip package if condition is defined and is false
+        next if defined($packages{$package}->{condition}) && !$packages{$package}->{condition}->();
         my $args = $packages{$package}->{installed} ? '--installed-only' : '--not-installed-only';
-        assert_script_run("zypper se -n $args --match-exact --details $package | grep " . $packages{$package}->{repo});
+        # Set flag if availability condition is defined and is true or not defined
+        my $available = !!(!defined($packages{$package}->{available}) || $packages{$package}->{available}->());
+        # Negate condition if package should not be available
+        my $cmd = $available ? '' : '! ';
+        $cmd .= "zypper se -n $args --match-exact --details $package";
+        # Verify repo only if package expected to be available
+        $cmd .= ' | grep ' . $packages{$package}->{repo} if $available;
+        # Record error in case non-zero return code
+        if (script_run($cmd)) {
+            if ($available) {
+                $errors .= "$package not found in @{ [ $packages{$package}->{repo} ] } or not preinstalled."
+                  . " Expected to be installed: @{ [ $packages{$package}->{installed} ? 'true' : 'false' ] }\n";
+            }
+            else {
+                $errors .= "$package found in @{ [ $packages{$package}->{repo} ] } repo, expected to be not available\n";
```

Comment:
Could you add single ticks around `$package` so that the separation between the variable content and the description string is clear? Also I recommend to mention the word "package" in before

Reply:
Makes sense, thanks!